Improvements to the port of crm_claim_rma

LeartS opened the pull request.

mistotebe commented on crm_claim_rma/models/crm_claim_rma.py:
```diff
@@ -251,19 +245,15 @@ def _warranty_limit_values(self, invoice, claim_type, product, claim_date):
 
     def set_warranty_limit(self):
```
Use @api.one or self.ensure_one()?

mistotebe commented on crm_claim_rma/models/crm_claim_rma.py:
```diff
@@ -371,10 +360,10 @@ def _warranty_return_address_values(self, product, company, warehouse):
-                'location_dest_id': location_dest_id}
+                'location_dest_id': location_dest.id}
 
     def set_warranty_return_address(self):
```
Use @api.one or self.ensure_one()?

mistotebe commented on crm_claim_rma/wizards/claim_make_picking.py:
```diff
@@ -0,0 +1,235 @@
+        if picking_type == 'in' and partner_id:
+            # Add the case of return to supplier !
+            lines = self._default_claim_line_ids()
+            return self._get_common_dest_location_from_line(lines)
```
Does this mean you sometimes return an id and sometimes a browse object? Maybe worth making the `_get_common_...` functions return a browse object?

LeartS commented:
@mistotebe I should have addressed your remarks, thanks.

The PR is ready for review. Keep in mind that I refactored and cleaned up the code without really knowing what it's supposed to do functionally, so I refactored while keeping the code the same functionally.
I cherry-picked some of the commits and changes mentioned here: https://github.com/OCA/rma/pull/32#issuecomment-127954640 and authored / rewrote others.

nhomar commented:
@aristobulo-eezee can you merge in order to make the review properly in the main pr to oca/rma. 

Thanks a lot to both of you......

nhomar commented:
wao.... that was fast
jejejeeje

LeartS commented:
Yeah, it's better to review there, thanks a lot. Very efficient community :)
